### ISSUE TITLE
PDAL is trashing GeoTIFF keys?

### ISSUE THREAD

**Issue by hobu** [PDAL/PDAL#1040]
`pdal info`:

```
PROJCS["NAD83 / UTM zone 18N",GEOGCS["NAD83",DATUM["North_American_Datum_1983",SPHEROID["unretrievable - using WGS84",6378137,298.257223563,AUTHORITY["EPSG","7019"]],AUTHORITY["EPSG","6269"]],PRIMEM["Greenwich",0],UNIT["degree",0.0174532925199433],AUTHORITY["EPSG","4269"]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",0],PARAMETER["scale_factor",0],PARAMETER["false_easting",0],PARAMETER["false_northing",0],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AUTHORITY["EPSG","26918"]]
```

lasinfo:

```
PROJCS["NAD83 / UTM zone 18N",
    GEOGCS["NAD83",
        DATUM["North_American_Datum_1983",
            SPHEROID["GRS 1980",6378137,298.2572221010002,
                AUTHORITY["EPSG","7019"]],
            AUTHORITY["EPSG","6269"]],
        PRIMEM["Greenwich",0],
        UNIT["degree",0.0174532925199433],
        AUTHORITY["EPSG","4269"]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",0],
    PARAMETER["central_meridian",-75],
    PARAMETER["scale_factor",0.9996],
    PARAMETER["false_easting",500000],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AUTHORITY["EPSG","26918"]]
```

Note the SPHEROID parameter. This is properly "GRS 1980" as coming from lasinfo, but a broken "unretrievable - using WGS84" from PDAL. I don't know that PDAL is trashing the key, but something is getting lost.

**Comment by abellgithub** [comment]
Mike gave pointed me to the file 18TWL865000.las.laz and it doesn't recreate the problem.  Can someone post a link to a file that causes the issue, cut down to only a few points?

**Comment by msmitherdc** [comment]
```
(production)gridusr@gridvm-puppy (issues/1922-celery-forest *) ~/GRiD: pdal --debug
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
PDAL debug information
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Version information
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
(1.0.1 (git-version: a82d94))

Debug build status
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Release

Enabled libraries
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

GEOS (3.3.2) - http://trac.osgeo.org/geos
GDAL (1.11.3) - http://www.gdal.org
LASzip (2.0.2) - http://laszip.org
libxml (2.7.6) - http://www.xmlsoft.org/
libgeotiff (1410) - http://trac.osgeo.org/geotiff

pdal info --metadata /u02/LASdata/hobudata/ct/utm18/18TBF5182.laz

    "spatialreference": "PROJCS[\"NAD83 / UTM zone 18N\",GEOGCS[\"NAD83\",DATUM[\"North_American_Datum_1983\",SPHEROID[\"unretrievable - using WGS84\",6378137,298.257223563,AUTHORITY[\"EPSG\",\"7019\"]],AUTHORITY[\"EPSG\",\"6269\"]],PRIMEM[\"Greenwich\",0],UNIT[\"degree\",0.0174532925199433],AUTHORITY[\"EPSG\",\"4269\"]],PROJECTION[\"Transverse_Mercator\"],PARAMETER[\"latitude_of_origin\",0],PARAMETER[\"central_meridian\",0],PARAMETER[\"scale_factor\",0],PARAMETER[\"false_easting\",0],PARAMETER[\"false_northing\",0],UNIT[\"metre\",1,AUTHORITY[\"EPSG\",\"9001\"]],AUTHORITY[\"EPSG\",\"26918\"]]",
```

so i still get the issue even with the updated libs (geotiff 1.4.1, proj-4.9.1, gdal 1.11.3)

**Comment by abellgithub** [comment]
That “unretrievable” bit is coming directly from the file.  It has a WKT VLR that we prefer over the GEOTIFF VLR.  You can do `strings 18TBF5182.laz` and search for COMPD_CS or whatever and you’ll see it.  I bet that lasinfo is ignoring the WKT VLR and just looking at the GEOTIFF VLR.  When I do that, I get good output.   In the end, your data is bad.

**Comment by abellgithub** [comment]
The file is 1.2, so perhaps we should do something different.  The spec does say that it's illegal to have both WKT and GeoTiff keys in the same file.

**Comment by hobu** [comment]
Let's change to the following:

1) PDAL ignores any wkt or `liblas` wkt VLRs for any LAS data <= 1.3
2) PDAL follows the LAS 1.4 WKT vs GeoTIFF rules

**Comment by abellgithub** [comment]
Yes.  I've fixed things to follow the spec more closely.  Not quite done.

**Comment by abellgithub** [comment]
Fixed via 129b574a075b57d12e82bdfbc1f596598038c0b2
